### ISSUE TITLE
NGINX + PHP-FPM net::ERR_INCOMPLETE_CHUNKED_ENCODING

### ISSUE THREAD

**Issue by apexJCL** [Microsoft/BashOnWindows#2100]
**Operating System Information:**

Windows Build Number: 10.0.15063 (15063.296)
Windows Version: Windows 10 Home Single Language x64
Version: 1703

**Issue:**

Running an NGINX instance with PHP-FPM enabled gives a timeout and an error on console `net::ERR_INCOMPLETE_CHUNKED_ENCODING`

**Steps to reproduce:**

1. Enable/Install WLS
2. Update/Upgrade (`apt update && apt upgrade`)
3. Install NGINX
4. Install php-fpm
5. Configure NGINX to use `php7.0-fpm`socket and support `index.php` file
6. Delete default `index` and create a new like following:

index.php
```php
<?php
phpinfo();
```

**Behavior:**

After enabling both services (`service nginx start` and `service php7.0-fpm start`), entering address 127.0.0.1 or **localhost** renders the phpinfo() but keeps waiting for a request finish. Secods later, on console an error will appear that reads `net::ERR_INCOMPLETE_CHUNKED_ENCODING`

Accesing the default `index.html` works alright, so I assume it all relies on `php-fpm` side

**Comment by Line117** [comment]
Check out my option
https://github.com/Microsoft/BashOnWindows/issues/1851#issuecomment-294299975

**Comment by apexJCL** [comment]
@Line117 I already checked your answer, here's what I did:

**TL:DR**
Adding `fastcgi_buffering off;` as explained [here](https://github.com/Microsoft/BashOnWindows/issues/1851#issuecomment-294299975) did the trick, now pages are loading fast and smoothly.

1. Install NGINX from stable releases (the same as you)
2. From here on, doing this changes the default user, because installing from Ubuntu repos gives you `www-data` user, whereas installing from NGINX repos gives you `nginx` user.
3. Install `php-fpm`
4. In the `nginx.conf` file, added on the `http` block the line `fastcgi_buffering off;`.
5. Restart `nginx` and `php7.0-fpm` service


I'd take a stake (blindly) and assume that the buffering issue it's due to disk access/filesystem related. I guess I can close this issue.

**Comment by kojinobumoto** [comment]
After adding `fastcgi_buffering off;` I've encountered the fastcgi_cache does not work (fastcgi became not to save any cache file).
As being described in [nginx's manual](http://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_buffering), nginx seemed not try to read the response from the FastCGI server.
So, I've increased the fastcgi_buffer_size.
```
fastcgi_buffer_size 128k;
fastcgi_buffers 4 256k;

```
Now, the `net::ERR_INCOMPLETE_CHUNKED_ENCODING` had disappeared and fastcgi_cache works.

**Comment by DrJLT** [comment]
> After adding `fastcgi_buffering off;` I've encountered the fastcgi_cache does not work (fastcgi became not to save any cache file).
> As being described in [nginx's manual](http://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_buffering), nginx seemed not try to read the response from the FastCGI server.
> So, I've increased the fastcgi_buffer_size.
> 
> ```
> fastcgi_buffer_size 128k;
> fastcgi_buffers 4 256k;
> ```
> 
> Now, the `net::ERR_INCOMPLETE_CHUNKED_ENCODING` had disappeared and fastcgi_cache works.

I'm running into a slightly different issue: about 2/5 of the requests hangs, and will eventually load if I keep refreshing.

Using
```
fastcgi_buffering off;
```
solved the problem.

I also tried your suggesting with fastcgi_buffer_size, but it didn't help for me. At least in my environment, it seems that the bug is with this module.
